plugins/derive/derive.ml

```diff
@@ -101,8 +101,8 @@ let start_deriving f suchthat lemma =
       in
 
   let terminator = Proof_global.make_terminator terminator in
-  let pstate = Proof_global.start_dependent_proof ~ontop:None lemma kind goals terminator in
-  Proof_global.simple_with_current_proof begin fun _ p ->
+  let pstate = Proof_global.start_dependent_proof lemma kind goals terminator in
+  Proof_global.modify_proof begin fun p ->
```

Comment:
@SkySkimmer note that there is a change of semantics here, due to `simple_with_current_proof` using the newline tactic.

Reply:
That's incorrect, it only passes the endline tactic (wrapped up in some genard stuff), that's the `_` in the before of the diff.

Reply:
Oh indeed, I got confused.